framework/src/executioners/FEProblemSolve.C

```diff
@@ -435,8 +432,17 @@ FEProblemSolve::solve()
           _console << COLOR_GREEN << solve_name << " Skipped!" << COLOR_DEFAULT << std::endl;
       }
 
+      // Assess convergence of the multi-system fixed point iteration
       if (!_using_multi_sys_fp_iterations)
         converged = true;
+      else
+      {
+        converged = _multi_sys_fp_convergence->checkConvergence(num_fp_multisys_iters) ==
+                    Convergence::MooseConvergenceStatus::CONVERGED;
+        if (_multi_sys_fp_convergence->checkConvergence(num_fp_multisys_iters) ==
+            Convergence::MooseConvergenceStatus::DIVERGED)
+          break;
+      }
```

Comment:
thanks for this cleanup

Reply:
that ended up being a fix too (diverged case) so thanks for the comment!